test/test-visually/demo.ts

```
@@ -0,0 +1,287 @@
+    private validateEmail(email: string): boolean {
+        return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
+    }
+    
+    validate(): boolean {
+        return !!(this.id && this.name && this.validateEmail(this._email));
+    }
+}
+
+// Decorators
+function Logger(prefix: string) {
+    return function <T extends { new(...args: any[]): {} }>(constructor: T) {
```

Biome (2.1.2)

[error] Do not shadow the global "constructor" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)